integrations/watsonx/src/haystack_integrations/components/embedders/watsonx/document_embedder.py

```diff
@@ -0,0 +1,204 @@
+# SPDX-FileCopyrightText: 2025-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from typing import Any
+
+from haystack import Document, component, default_from_dict, default_to_dict
+from haystack.utils import Secret, deserialize_secrets_inplace
+from ibm_watsonx_ai import Credentials
+from ibm_watsonx_ai.foundation_models import Embeddings
+
+
+@component
+class WatsonXDocumentEmbedder:
+    """
+    Computes document embeddings using IBM watsonx.ai models.
+
+    ### Usage example
+
+    ```python
+    from haystack import Document
+    from haystack_integrations.components.embedders.watsonx.document_embedder import (
+        WatsonXDocumentEmbedder,
+    )
+
+    documents = [
+        Document(content="I love pizza!"),
+        Document(content="Pasta is great too"),
+    ]
+
+    document_embedder = WatsonXDocumentEmbedder(
+        model="ibm/slate-30m-english-rtrvr",
+        api_key=Secret.from_env_var("WATSONX_API_KEY"),
+        url="https://us-south.ml.cloud.ibm.com",
+        project_id=Secret.from_env_var("WATSONX_PROJECT_ID"),
+    )
+
+    result = document_embedder.run(documents=documents)
+    print(result["documents"][0].embedding)
+
+    # [0.017020374536514282, -0.023255806416273117, ...]
+    ```
+    """
+
+    def __init__(
+        self,
+        *,
+        model: str = "ibm/slate-30m-english-rtrvr",
+        api_key: Secret | None = None,
+        url: str = "https://us-south.ml.cloud.ibm.com",
+        project_id: Secret | None = None,
```

Comment:
In relation to this comment https://github.com/deepset-ai/haystack-core-integrations/pull/1949/files#r2174395073

```suggestion
        project_id: Secret = Secret.from_env_var("WATSONX_PROJECT_ID"),
```

If this looks good to you let's make the same change for the other components.